code/controllers/subsystem/objects.dm

```diff
@@ -20,7 +20,11 @@ var/list/processing_objects = list()
 /datum/subsystem/obj/Initialize()
 	for(var/atom/object in world)
 		if(!(flags & ATOM_INITIALIZED))
+			var/time_start = world.timeofday
 			object.initialize()
+			var/time = (world.timeofday - time_start)
+			if(time > 1 SECONDS)
+				log_debug("Slow object initialize. [object] ([object.type]) at [object.x],[object.y],[object.z] took [time] seconds to initialize.")
```

Comment:
Oh, right. You might want to take into account the scenario that the object is inside another object, in which case `x`, `y`, and `z` will tell you nothing

Reply:
Can't byond handle it somehow? Adding that logic here is adding unreadable shitcode

Reply:
just `get_turf` will probably tell you enough? get the coordinates of the turf instead of the object's.

Reply:
also, the profiler can be started server-side now, and the data can be dumped to json, so maybe you could look into that instead of doing this manually. I know TG already implemented that. I think they start the profiler as soon as the world loads.

Reply:
I was hoping for something more straightforward honestly

Reply:
by server-side profiler you mean extools?

Reply:
No, he means the BYOND profiler built into BYOND

Reply:
I think this PR is fine as is, it can be used to support shifty's theory or to discard it altogether. I might look into something more sophisticated in the future if the need arises.

If you don't think this PR is needed at all feel free to close it, no strings attached.